src/Domain/CoordinateMaps/TimeDependent/SphericalCompression.cpp

```diff
@@ -0,0 +1,385 @@
+}
+
+// Evaluate \f$\lambda_{00}^{\prime}(t) / sqrt{4\pi}\f$.
+double dt_lambda00_y00(
+    const std::string& f_of_t_name, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) noexcept {
+  ASSERT(functions_of_time.find(f_of_t_name) != functions_of_time.end(),
+         "Could not find function of time: '"
+             << f_of_t_name << "' in functions of time. Known functions are "
+             << keys_of(functions_of_time));
+  return functions_of_time.at(f_of_t_name)->func_and_deriv(time)[1][0] * 0.25 *
+         M_2_SQRTPI;
+}
+
+// Evaluate \f$\rho^i = \xi^i - C^i\f$ or \f$r^i = x^i - C^i\f$.
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 3> radial_position(
+    const std::array<T, 3>& coords,
+    const std::array<double, 3>& center) noexcept {
+  std::array<tt::remove_cvref_wrap_t<T>, 3> result{};
+  for (size_t i = 0; i < 3; ++i) {
+    gsl::at(result, i) = gsl::at(coords, i) - gsl::at(center, i);
+  }
+  return result;
+}
+
+// Adds the correction (terms other than the source coordinates) to
+// compute either the mapped coordinate (if input is initially the source
+// coordinates and and lambda_y is lambda00'(t) /sqrt(4*M_PI)) or the frame
+// velocity (if input is initially zero and lambda_y is lambda00'(t)
+// /sqrt(4*M_PI)).
+template <typename T>
+void correct_mapped_coordinate_or_frame_velocity(
+    gsl::not_null<tt::remove_cvref_wrap_t<T>*> input,
+    const T& source_radial_coord, const T& source_radius, const double lambda_y,
+    const double min_radius, const double max_radius) noexcept {
+  if (source_radius < min_radius) {
+    *input -= lambda_y * source_radial_coord / min_radius;
+  } else if (source_radius < max_radius) {
+    *input -= lambda_y * source_radial_coord *
+              (max_radius / source_radius - 1.0) / (max_radius - min_radius);
+  }
+}
+
+// This function computes the (i,j) component of the Jacobian matrix
+// \f$\partial x^i / \partial \xi^j.\f$
+template <typename T>
+void jacobian_component(gsl::not_null<tt::remove_cvref_wrap_t<T>*> input,
+                        const std::array<T, 3>& source_radial_position,
+                        const T& source_radius, const double lambda_y,
+                        const double min_radius, const double max_radius,
+                        const size_t i, const size_t j) noexcept {
+  if (source_radius >= min_radius) {
+    if (source_radius <= max_radius) {
+      *input = gsl::at(source_radial_position, i) *
+               gsl::at(source_radial_position, j) * lambda_y * (max_radius) /
+               (max_radius - min_radius) / cube(source_radius);
+      if (i == j) {
+        *input += (1.0 - lambda_y * (max_radius / source_radius - 1.0) /
+                             (max_radius - min_radius));
+      }
+    } else {
+      if (i == j) {
+        *input = 1.0;
+      } else {
+        *input = 0.0;
+      }
+    }
```

Comment:
Do we want to Error in this case?  A Block with rho >= rho_max shouldn't even contain a SphericalCompression map.